code/src/components/PostMessage.js

```diff
@@ -0,0 +1,64 @@
+import React, { useState, useEffect } from 'react';
+
+const PostMessage = ({ newMessage, fetchPosts }) => {
+  const [newPost, setNewPost] = useState(''); // Initial state is an empty string
+  const [errorMessage, setErrorMessage] = useState(''); // Initial state is an empty string
+
+  useEffect(() => {
+    if (newPost.length >= 141) {
+      setErrorMessage('Your message is too long 😔');
+    } else {
+      setErrorMessage('');
+    }
+  }, [newPost])
+
+  // This function handles the API with method POST
+  // It is called when the user submits the form below, clicking the submit button
+  const handleFormSubmit = (event) => {
+    event.preventDefault();
+    console.log('newPost onformsubmit:', newPost);
+    if (newPost.length <= 4) {
+      setErrorMessage('Your message is too short, it needs at least 5 letters 😔');
+    } else {
+      const options = {
+        method: 'POST',
+        body: JSON.stringify({
+          message: `${newPost}`
+        }),
+        headers: { 'Content-Type': 'application/json' }
+      };
+
+      // console.log('options:', options);
+      fetch('https://happy-thoughts-ux7hkzgmwa-uc.a.run.app/thoughts', options)
+        .then((response) => response.json())
+        .then((data) => {
+          newMessage(data)
+          setNewPost('')
+          fetchPosts()
```

Comment:
Interesting using ```fetchPosts``` as a deconstructing prop! Makes sense to re-render the API here.